analysisservice/src/main/java/nu/yona/server/analysis/service/AnalysisEngineService.java

```diff
@@ -246,9 +252,40 @@ private void addOrUpdateDayTruncatedActivity(UserAnonymizedEntityHolder userAnon
 		{
 			return Optional.empty();
 		}
-		return Optional.ofNullable(activityRepository.findOverlappingOfSameApp(dayActivity.get(), payload.deviceAnonymizedId,
+
+		List<Activity> overlappingOfSameApp = activityRepository.findOverlappingOfSameApp(dayActivity.get(), payload.deviceAnonymizedId,
 				matchingGoal.getActivityCategoryId(), payload.application.orElse(null), payload.startTime.toLocalDateTime(),
-				payload.endTime.toLocalDateTime()));
+				payload.endTime.toLocalDateTime());
+
+		// The prognosis is that there is no or one overlapping activity of the same app,
+		// because we don't expect the mobile app to post app activity that spans other existing same app activity
+		// (that indicates that there is something wrong in the app activity bookkeeping).
+		// Network activity (having payload.application == null) also is not expected to exist and overlap, as network activity
+		// has a very short time span.
+		// So log if there are multiple overlaps.
+
+		if (overlappingOfSameApp.isEmpty())
+		{
+			return Optional.empty();
+		}
+		else if (overlappingOfSameApp.size() == 1)
+		{
+			return Optional.of(overlappingOfSameApp.get(0));
+		}
+		else
+		{
+			String relatedTo = payload.application.isPresent()
+					? MessageFormat.format("same app ''{0}'' activity", payload.application.get()) : "network activity";
```

Comment:
This now results in messages like this:
* ``Multiple overlapping activities of same app 'abc' activity found.``
* ``Multiple overlapping activities of network activity found.``

Can't you change it this way?
* ``Multiple overlapping app activities of 'abc' found.``
* ``Multiple overlapping network activities found.``

That would be more clear to me.

Reply:
Good point, changed